test/integration/tools/system/package_manager_test.py

```diff
@@ -315,9 +398,37 @@ def fake_check(*args, **kwargs):
     assert tool._conanfile.command == result
 
 
+@pytest.mark.parametrize("tool_class, result", [
+    # cross-compile but arch_names=None -> do not add host architecture
+    # https://github.com/conan-io/conan/issues/12320 because the package is archless
+    (Apt, 'apt-get install -y --no-install-recommends package1=0.1 package2=0.2'),
+    (Yum, 'yum install -y package1-0.1 package2-0.2'),
+    (Dnf, 'dnf install -y package1-0.1 package2-0.2'),
+    (PacMan, 'pacman -S --noconfirm package1 package2'),
+])
+def test_tools_install_archless_with_version(tool_class, result):
+    conanfile = ConanFileMock()
+    conanfile.settings = MockSettings({"arch": "x86"})
+    conanfile.settings_build = MockSettings({"arch": "x86_64"})
+    conanfile.conf.define("tools.system.package_manager:tool", tool_class.tool_name)
+    conanfile.conf.define("tools.system.package_manager:mode", "install")
+    with mock.patch('conan.ConanFile.context', new_callable=PropertyMock) as context_mock:
+        context_mock.return_value = "host"
+        tool = tool_class(conanfile, arch_names={})
+
+        def fake_check(*args, **kwargs):
+            return ["package1=0.1", "package2=0.2"]
+        from conan.tools.system.package_manager import _SystemPackageManagerTool
+        with patch.object(_SystemPackageManagerTool, 'check', MagicMock(side_effect=fake_check)):
+            tool.install(["package1=0.1", "package2=0.2"])
+
+    assert tool._conanfile.command == result
+
+
+
 @pytest.mark.parametrize("tool_class, result", [
     (Apk, 'apk info -e package'),
-    (Apt, 'dpkg-query -W -f=\'${Status}\' package | grep -q "ok installed"'),
+    (Apt, 'dpkg -l | grep -q -E "^ii\\s+\\bpackage\\s+\\b"'),
```

Comment:
Curiosity, but what is the reasoning behind this change? a regular expression is a tad harder to parse - 

the code as it was was added to address an issue 
https://github.com/conan-io/conan/pull/3033
https://github.com/conan-io/conan/issues/3032

just want to make sure we don't introduce regressions

also note that you can do for example `dpkg-query -W -f='${version}' libudev-dev` to get the version of a package
it can be combined with other things like `${Status}`

changes are dpkg -l is fine, but may need to consider whether it causes performance issues due to repeated calls on systems that have many packages installed (`dpkg -l` gives you the list of all packages, dpkg-query can be tailored to output info of a specific package